src/hooks/useFormFields.ts

```diff
@@ -0,0 +1,32 @@
+import React, { useState } from "react";
+import DefaultFieldKey from "../constants/DefaultFieldKey";
+import StudentRole from "../constants/StudentRole";
+import { DefaultField, DynamicField } from "../types";
+
+export type DefaultFormField = DefaultField & { role: StudentRole };
```

Comment:
What does the `&` operator do here?

Reply:
good q! this is an [intersection type](https://www.typescriptlang.org/docs/handbook/2/objects.html#intersection-types), which lets us extend the `DefaultField` type and add an additional `role` property